p2p/5.Create_an_ERC-20/README.md

```diff
@@ -0,0 +1,245 @@
+
+### 📌 **Tasks**:
+
+- Create in the `ERC20` contract the following variables:
+    - All variables except `_totalSupply` will be defined on declaration
+    - `name` which corresponds to the name of the token
+    - `symbol` which corresponds to the abbreviation of the token name
+    - `decimals` which corresponds to the number of decimals that the token can have
+    - `_totalSupply` which corresponds to the total token created
+> ⚠️ Take care about the [visibility](https://docs.soliditylang.org/en/v0.8.21/contracts.html#state-variable-visibility) of your variables! The names of the variables can give you clues 😉
+
+- Create in the `ERC20` contract the following [mappings](https://docs.soliditylang.org/en/v0.8.21/types.html#mapping-types):
+    - `_balances` wich corresponds to the balance of all wallets
+    - `_allowances` which corresponds to the number of tokens whose owner has authorized the spender to use
+> ⚠️ Take care about the [visibility](https://docs.soliditylang.org/en/v0.8.21/contracts.html#state-variable-visibility) of your mappings!
+
+- Initialize `_totalSupply` in the [constructor](https://docs.soliditylang.org/en/v0.8.21/contracts.html#constructor) of the contract.
+> ⚠️ Don't forget to assign `_totalSupply` to a wallet and to emit an event
+
+### 📚 **Documentation**:
+
+- [Variable visibility](https://docs.soliditylang.org/en/v0.8.21/contracts.html#state-variable-visibility)
+- [Mapping](https://docs.soliditylang.org/en/v0.8.21/types.html#mapping-types)
+- [Constructor](https://docs.soliditylang.org/en/v0.8.21/contracts.html#constructor)
+
+## Step 3 - Let's get into the hard
+
+### 📑 **Description**:
+
+In this step, you will do the most important part: the implementation of the standard functions.
+
+### 📌 **Tasks**:
+
+- Implement in the `ERC20` contract all the functions present in the `IERC20` interface.
+    - You can copy function prototypes directly into the contract.
+
+> ⚠️ Don't forget to add the `override` modifier
+
+### 👨‍🏫 **Advices**:
+
+- Read the doc of the [ERC-20: Token Standard](https://eips.ethereum.org/EIPS/eip-20), it will help you to understand the purpose of the different functions.
+- Take care about the [visibility](https://docs.soliditylang.org/en/v0.8.21/contracts.html#state-variable-visibility) of your functions!
+- Use if into revert or require, to handle [error cases](https://docs.soliditylang.org/en/v0.8.21/contracts.html#errors-and-the-revert-statement).
+
+### 📚 **Documentation**:
+
+- [ERC-20: Token Standard](https://eips.ethereum.org/EIPS/eip-20)
+- [Function visibility](https://docs.soliditylang.org/en/v0.8.21/contracts.html#function-visibility)
+- [Error cases](https://docs.soliditylang.org/en/v0.8.21/contracts.html#errors-and-the-revert-statement)
+
+## Step 4 - Let's test it
+
+### 📑 **Description**:
+
+In this step, you will test, with foundry, the functions that you have implemented previously.
```

Comment:
same again, you can add a few phrases to explain why tests are important in smart contracts development even more than in other areas, what is foundry....